sel/strongly_ordered.hpp

```diff
@@ -0,0 +1,15 @@
+#pragma once
+
+#include <compare>
+#include <concepts>
+
+namespace sel {
+
+template <class T>
+concept strongly_ordered =       //
+    std::totally_ordered<T> and  //
+    requires (T x) {
+      { x <=> x } -> std::same_as<std::strong_ordering>;
+    };
+
+}  // namespace sel
```

Comment:
the syntax is mind blowing

Reply:
Mind blowingly cool? It is to me -- coming from all the older C++ versions.

Reply:
yeah I haven't seen the crazy templating tricks (enable_if, sfinae, ...) yet so if this replaces that sort of thing its mind blowingly cool